fastly/compute_acl_test.go

```diff
@@ -0,0 +1,275 @@
+package fastly
+
+import (
+	"encoding/json"
+	"net"
+	"strings"
+	"testing"
+)
+
+func TestClient_ComputeACL(t *testing.T) {
+	t.Parallel()
+
+	const computeACLTestName = "test-compute-acl"
+
+	var computeACLsListResp *ListComputeACLsResponse
+	var err error
+
+	// List all compute ACLs.
+	Record(t, "compute_acls/list_acls", func(c *Client) {
+		computeACLsListResp, err = c.ListComputeACLs()
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Make sure the test compute acl we're going to create isn't among them.
+	for _, acl := range computeACLsListResp.Data {
+		if acl.Name == computeACLTestName {
+			t.Errorf("found test compute ACL %q, aborting", computeACLTestName)
+		}
+	}
+
+	// Create a compute ACL for testing.
+	var acl *ComputeACL
+	Record(t, "compute_acls/create_acl", func(c *Client) {
+		acl, err = c.CreateComputeACL(&CreateComputeACLInput{
+			Name: computeACLTestName,
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if acl.Name != computeACLTestName {
+		t.Errorf("unexpected acl name: got %q, expected %q", acl.Name, computeACLTestName)
+	}
+
+	// Ensure we delete the test compute ACL at the end.
+	defer func() {
+		Record(t, "compute_acls/cleanup", func(c *Client) {
+			err = c.DeleteComputeACL(&DeleteComputeACLInput{
+				ComputeACLID: acl.ComputeACLID,
+			})
+		})
+		if err != nil {
+			t.Errorf("error during compute ACL cleanup: %v", err)
+		}
+	}()
+
+	// Describe the test compute ACL.
+	var describeComputeACLResponse *ComputeACL
+	Record(t, "compute_acls/describe_acl", func(c *Client) {
+		describeComputeACLResponse, err = c.DescribeComputeACL(&DescribeComputeACLInput{
+			ComputeACLID: acl.ComputeACLID,
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if describeComputeACLResponse.Name != acl.Name {
+		t.Errorf("unexpected acl name: got %q, expected %q", describeComputeACLResponse.Name, acl.Name)
+	}
+
+	if describeComputeACLResponse.ComputeACLID != acl.ComputeACLID {
+		t.Errorf("unexpected acl ID: got %q, expected %q", describeComputeACLResponse.ComputeACLID, acl.ComputeACLID)
+	}
+
+	// Add a bunch of entries to the test compute ACL.
+	batchCreateEntries := struct {
```

Comment:
Since the shape of this JSON object is defined by the API itself, I'm wondering whether we should just make it part of the go-fastly wrapper for the API, instead of just accepting raw JSON in `BatchModifyComputeACLEntries`. That would require the CLI to unmarshal the user's provided JSON content into the structure before passing it to go-fastly, but that could be reasonable as it means unmarshaling failures would be caught there instead of by the API itself.

Reply:
Addressed in 542f3a33d9b4fe9575d229819a5ee09042ce2647.